src/common/config_cacher.h

```diff
@@ -53,6 +53,10 @@ class md_config_cacher_t : public md_config_obs_t {
   operator ValueT() const {
     return value_cache.load();
   }
+
+  ValueT operator()() const {
```

Comment:
Do we really need the user conversion operator after introducing `operator()`? I believe backward compatibility isn't an issue as the cacher has just two usages (three with https://github.com/ceph/ceph/pull/60610) across the entire code base.

Reply:
Makes sense to remove it. I'll add commits to modify the users and to delete that interface
(like I mentioned above - I suspect it was always meant to be operator()())

Reply:
@rzarzynski - done

Reply:
I don't find osd_scrub_chunk_min() to be an intuitive interface.  I'd actually prefer to keep the conversion function (```operator ValueT() const```) and add a named ```ValueT to_value() const``` function.  I'm ok with being overruled if you two find this to be clearer.

Reply:
I think it is much clearer than the cast (which has a surprising behavior), and is the most elegant.
We can probably use '*' (which is common enough, e.g. optional or expected), but in our context - just
a wrapper around a value - I like the '()'.

Reply:
I prefer a single method approach.
Whether `operator*` or `operator()` isn't so important. I guess `*my_cached_configurable` would look slightly more familiar but the feeling isn't strong enough to block and request changes.